src/System.Private.Interop/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -125,16 +125,16 @@ public static int SizeOf<T>()
 
         public static int SizeOf<T>(T structure)
         {
-            return SizeOf<T>();
+            return SizeOf((object)structure);
```

Comment:
This change is to match the CoreCLR behavior in cases when `structure` is not of the exact type `T`.